### PR TITLE
WIP: Migrate javax -> jakarta (Jakarta EE 9)

### PR DESCRIPTION
**Pull request by erikgb**
GitHub issue resolved: # 

Briefly describe the changes proposed in this PR:

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md) for more details):

 - [ ] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [ ] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [ ] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [ ] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

File: .github/workflows/develop-status.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        jdk: [11, 19]
+        jdk: [17, 19]
```
Comment (jeenbroekstra) [inline .github/workflows/develop-status.yml:14]: I don't think that's a migration we can do at this point - there's significant subset of our users that require support for Java 11.

Comment (erikgb) [reply .github/workflows/develop-status.yml:14]: I see and can understand why. 😃 Could it be an option to require Java 17 to use the modules with spring/spring-boot dependencies - if I can find a way to configure it?

Comment (JervenBolleman) [reply .github/workflows/develop-status.yml:14]: With RDF4j V5 we can all go to java 17 IMHO.  Maybe it is better I open that in a discussion topic though.

Comment (erikgb) [reply .github/workflows/develop-status.yml:14]: > With RDF4j V5 we can all go to java 17 IMHO. Maybe it is better I open that in a discussion topic though.

I hope so. I can start a discussion about this. It is a requirement for the javax -> jakarta migration that blocks a lot of other upgrades like servlet containers (Jetty, Tomcat), Spring/Spring Boot+++

Comment (jetztgradnet) [reply .github/workflows/develop-status.yml:14]: why would you need to bump this to Java 17? RDF4J compiled for Java 11 can still be *used* with Java 17, no?

Comment (erikgb) [reply .github/workflows/develop-status.yml:14]: Spring Boot 3.x and Spring 6.x requires Java 17. And without upgrading Spring we cannot migrate from javax to jakarta. This is WIP. We need to find a smart way of doing this....

Comment (jetztgradnet) [reply .github/workflows/develop-status.yml:14]: Sorry, I had a misunderstanding: I was thinking that some (external) Spring app using Java 17 would not require RDF4J also being compiled for Java 17. But you are actually talking about the RDF4J workbench using Spring inside, so RDF4J depending on Spring, not the other way round, right? 

So maybe we create a separate module that is using a Spring6/Spring Boot 3/Java 17 variant of the workbench?

Comment (erikgb) [reply .github/workflows/develop-status.yml:14]: The problem is the dependencies. You must choose, either javax or jakarta. Never both.

**Pull request closed without merging**